src/sql/parts/connection/connectionDialog/connectionDialogService.ts

```diff
@@ -76,6 +76,11 @@ export class ConnectionDialogService implements IConnectionDialogService {
 	private _connectionErrorTitle = localize('connectionError', 'Connection error');
 	private _dialogDeferredPromise: Deferred<IConnectionProfile>;
 
+	/**
+	 * This is used to work around the interconnectedness of this code
+	 */
+	private ignoreNextConnect = false;
```

Comment:
this seems like a hack.  We should at least expand the comment to explain the need for this workaround.  And ideally rework the code so it isn't needed..though I don't understand why it's needed yet.

Reply:
It really is a hack, the problem is this dialog is so tied in with the connection management service, that when I am trying to get credentials to expand in the azure tree, the connection management service adds it automatically to the normal server tree. This is the smallest change I could come up with. I would otherwise have to refactor this and the connection management service, or create another function in the connectionmanagement service in order to get the behavior I need.